### ISSUE TITLE
Evict expired HTTP connections opened by HttpClientHelper

### ISSUE THREAD

**Issue by s4nk** [gradle/gradle#6126]
### Expected Behavior
Improvement suggestion: HttpClient used in Gradle should cleanup connections that were closed on the server side. This could be default behavior or a configurable opt-in.

### Current Behavior
Currently Gradle Daemon may leave behind up to 20 half closed connections per host in CLOSE_WAIT state, if server side initiates closing a connection.

### Context
The problem is described in Apache HTTP Client documentation: https://hc.apache.org/httpcomponents-client-ga/tutorial/html/connmgmt.html#d5e418. The fix is a one-liner, `HttpClientBuilder` allows to enable eviction of expired connections from the connection pool by a background thread: 
```
builder.evictExpiredConnections();
```

### Your Environment
Any environment. Seen on Gradle 4.8.1 and 4.9.

**Comment by s4nk** [comment]
I can open a PR, if this is something that could be added. I'd just need to know if this should be configurable or always enabled.

**Comment by IainHull** [comment]
I am a colleague of @s4nk.

This issue seems to be triggering a network bug in docker when we run our builds inside docker containers with bridge networking. The result is a lot of half closed connections to our artifactory server which eventually causes connection timeouts downloading dependencies.

We have spent a number of weeks trying to track this issue down. We wonder if anyone else has seen connection issues using gradle from inside docker? Our build agents are quite large 24 cores, and often run 6 or more concurrent builds inside docker (which can result in 144 connections to artifactory).

We have tested a gradle change locally which seems to resolve the issues, and would love to get this verified and added to upstream.

This might not technically be a gradle bug, but it does help gradle to play nice with docker.
